system-test/managed_writer_client_test.ts

```diff
@@ -857,6 +865,280 @@ describe('managedwriter.WriterClient', () => {
+              throw err;
+            }
+            offset += 10;
+            connection.close(); // Close connection on every append to trigger reconnection
+          }
+
+          const res = await connection.finalize();
+          connection.close();
+          assert.equal(res?.rowCount, 500);
+
+          writer.close();
+        } finally {
+          client.close();
+        }
+      }).timeout(2 * 60 * 1000);
+    });
+
+    describe('should manage to send data in parallel', () => {
+      it('every 10 request drops the connection', async () => {
+        bqWriteClient.initialize();
+        const client = new WriterClient();
+        client.enableWriteRetries(true);
+        client.setMaxRetryAttempts(10);
+        client.setClient(bqWriteClient);
+
+        try {
+          const flakyTableId = generateUuid() + '_reconnect_on_close';
+          const [table] = await bigquery
+            .dataset(flakyDatasetId)
+            .createTable(flakyTableId, {
+              schema,
+              location: flakyRegion,
+            });
+          projectId = table.metadata.tableReference.projectId;
+          parent = `projects/${projectId}/datasets/${flakyDatasetId}/tables/${flakyTableId}`;
+
+          const connection = await client.createStreamConnection({
+            streamType: managedwriter.PendingStream,
+            destinationTable: parent,
+          });
+
+          const writer = new JSONWriter({
+            connection,
+            protoDescriptor,
+          });
+
+          const pendingWrites: PendingWrite[] = [];
+          const iterations = new Array(50).fill(1);
+          let offset = 0;
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          for (const _ of iterations) {
+            const rows = generateRows(10);
+            const pw = writer.appendRows(rows, offset);
+            pendingWrites.push(pw);
+            offset += 10;
+          }
+
+          await Promise.all(pendingWrites.map(pw => pw.getResult()));
+
+          const res = await connection.finalize();
+          connection.close();
+          assert.equal(res?.rowCount, 500);
+
+          writer.close();
+        } finally {
+          client.close();
+        }
+      }).timeout(2 * 60 * 1000);
+
+      it('every 10 request there is a in stream INTERNAL error', async () => {
+        bqWriteClient.initialize();
+        const client = new WriterClient();
+        client.enableWriteRetries(true);
+        client.setClient(bqWriteClient);
+
+        try {
+          const connection = await client.createStreamConnection({
+            streamType: managedwriter.PendingStream,
+            destinationTable: parent,
+          });
+
+          let numCalls = 0;
+          let numSucess = 0;
+          const conn = connection['_connection'] as gax.CancellableStream;
+          sandbox
+            .stub(conn, 'write')
+            .callsFake(
+              (
+                chunk: any,
```

Comment:
You can probably do `chunk: unknown` here, and the linter won't complain, and the `as` below should still work.